crates/storage/src/lazy/lazy_cell.rs

```diff
@@ -545,4 +545,63 @@ mod tests {
             Ok(())
         })
     }
+
+    #[test]
+    fn regression_test_for_issue_570() -> ink_env::Result<()> {
```

Comment:
The entire test can be heavily simplified. You only need to push a `(Option, i32)` tuple with values `(None, 1)` to the contract storage and then read it again in the next step and assert that the first value is `None` and the second value is `1`. Done.

Reply:
Or am I missing something?

Reply:
So that's a clever idea, I guess you thought the `1` would be written out to the first pointer and on the next `pull_spread` we would try to read a `Some(...)` there and fail. It doesn't work unfortunately because of this: `push_spread` for `Option` writes `is_some()` in the first ptr slot and increments the pointer. Hence on the next `pull_spread` we immediately return the `None` and continue pulling the `i32`.

Reply:
Using a tuple also doesn't simplify the test significantly, because for each tuple element the entire `push_spread` recursion is called. So storage is basically overwritten until I introduce `LazyCell` into the tuple.

So overall I'm quite happy with the simplicity of the test. But yeah lmk what you think.

I mean it could also be simplified to something like
```
assert_eq!(ptr_after_pushing_none, ptr_after_pushing_some);
assert_eq!(ptr_after_pulling_none, ptr_after_pulling_some);
``

Reply:
Thanks for the elaboration!